### PR TITLE
Account Ajax Integration Tests

### PR DESCRIPTION
**Pull request by crhallberg**
Testing configs, cache clearing, account icon display.

### CODE REVIEW

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+                'Catalog' => ['driver' => 'Demo'],
+                'Authentication' => [
+                    'enableAjax' => true,
+                    'enableDropdown' => false
+                ]
+            ]
+        ]);
+    }
+
+    protected function setJSStorage($states)
```
Comment (demiankatz) [inline module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php]: Do you mind adding function comments to the undocumented functions in here?

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+        // Clear different cache
+        $session->evaluateScript('VuFind.account.clearCache("holds");');
+        $cookies = $this->getJSStorage();
+        $this->assertEquals(true, $cookies['fines'] != null);
+        // Clear correct cache
+        $session->evaluateScript('VuFind.account.clearCache("fines");');
+        $cookies = $this->getJSStorage();
+        $this->assertEquals(true, $cookies['fines'] == null);
+
+        /* TODO: Not working in tests, but working in browser
```
Comment (demiankatz) [inline module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php]: Is this something you need help troubleshooting?

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+    protected function checkIcon($cookies, $checkClass)
+    {
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        foreach ($cookies as $cookie) {
+            $this->snooze();
+            $this->setJSStorage($cookie);
+            $session->reload();
+            $page = $session->getPage();
+            // error_log(print_r($this->getJSStorage(), true));
```
Comment (demiankatz) [inline module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php]: Can we delete this?

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php (hunk 2 above)
Comment (crhallberg) [reply module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php]: Sure! Each individual clearCache seems to work (ie. `clearCache("fines")`) but the universal one (no argument passed) wasn't working on my server during the test.

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php (hunk 1 above)
Comment (crhallberg) [reply module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php]: Will do

File: module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php (hunk 3 above)
Comment (crhallberg) [reply module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php]: Yes.

**Pull request merged**